drivers/base/bootvid/i386/pc/vga.c

```diff
@@ -199,80 +191,38 @@ DisplayCharacter(
 
 static VOID
 NTAPI
-SetPaletteEntryRGB(IN ULONG Id,
-                   IN ULONG Rgb)
+SetPaletteEntryRGB(
+    _In_ ULONG Id,
+    _In_ RGBQUAD Rgb)
 {
-    PCHAR Colors = (PCHAR)&Rgb;
-
     /* Set the palette index */
     __outpb(VGA_BASE_IO_PORT + DAC_ADDRESS_WRITE_PORT, (UCHAR)Id);
 
     /* Set RGB colors */
-    __outpb(VGA_BASE_IO_PORT + DAC_DATA_REG_PORT, Colors[2] >> 2);
-    __outpb(VGA_BASE_IO_PORT + DAC_DATA_REG_PORT, Colors[1] >> 2);
-    __outpb(VGA_BASE_IO_PORT + DAC_DATA_REG_PORT, Colors[0] >> 2);
+    __outpb(VGA_BASE_IO_PORT + DAC_DATA_REG_PORT, GetRValue(Rgb) >> 2);
+    __outpb(VGA_BASE_IO_PORT + DAC_DATA_REG_PORT, GetGValue(Rgb) >> 2);
+    __outpb(VGA_BASE_IO_PORT + DAC_DATA_REG_PORT, GetBValue(Rgb) >> 2);
 }
 
 VOID
 NTAPI
-InitPaletteWithTable(
+VidpInitPaletteWithTable(
```

Comment:
```suggestion
InitPaletteWithTable(
```
to emphasise it's a per-arch dependent global helper function and not a generic private bootvid helper.